tests/ci_build/build_r_pkg_with_cuda.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+set -e
+set -x
+
+if [ "$#" -ne 1 ]
+then
+  echo "Build the R package tarball with CUDA code. Usage: $0 [commit hash]"
+  exit 1
+fi
+
+commit_hash="$1"
+
+make Rpack
+mv xgboost/ xgboost_rpack/
+
+mkdir build
+cd build
+cmake .. -GNinja -DUSE_CUDA=ON -DR_LIB=ON
+ninja
+cd ..
+
+rm xgboost
+rm -v xgboost_rpack/configure
+rm -rfv xgboost_rpack/src
+mkdir -p xgboost_rpack/src
+cp -v lib/xgboost.so xgboost_rpack/src/
+echo 'all:' > xgboost_rpack/src/Makefile
+echo 'all:' > xgboost_rpack/src/Makefile.win
+mv xgboost_rpack/ xgboost/
```

Comment:
This super wacky hack is found in https://github.com/dmlc/xgboost/blob/master/cmake/RPackageInstall.cmake.in and https://github.com/dmlc/xgboost/blob/master/cmake/RPackageInstallTargetSetup.cmake . This hack lets us bypass the normal build process of R and have R use `xgboost.so` that we've already built.